cmd/juju/charmhub/download.go

```diff
@@ -55,27 +55,32 @@ See also:
 
 // NewDownloadCommand wraps downloadCommand with sane model settings.
 func NewDownloadCommand() cmd.Command {
-	return modelcmd.Wrap(&downloadCommand{},
-		modelcmd.WrapSkipModelInit,
-	)
+	return modelcmd.Wrap(&downloadCommand{
+		charmHubCommand: newCharmHubCommand(),
+		fallbackArch:    utilsarch.HostArch(),
```

Comment:
I think using the host arch as a fallback might be problematic here. What if I am running the cli on an arm64 machine and my controller only supports amd64?

Reply:
Specify an arch via `--arch=amd64`

Reply:
Perhaps then the `--arch` should be mandatory then?